### ISSUE TITLE
Export character from blender to character creator 4

### ISSUE THREAD

**Issue by drkl151** [soupday/cc_blender_tools#185]
Hello everyone, I have a difficult and most likely stupid question)
Maybe someone can suggest how to solve this problem.
I imported a character from Character Creator 4 into blender via a plugin, added my custom objects to it and now I want to import from blender back to CC4 via the same plugin, but it swears at armature (? 😅). I made all my custom objects children, but as I understand it, you need to somehow bind their bones, or to be honest, I don’t really understand what needs to be done to import the character without problems.
If you have any ideas please let me know :)

![image](https://github.com/soupday/cc_blender_tools/assets/81778290/b311c7c1-3156-4bda-87ba-7004998d1579)

**Comment by soupday** [comment]
Character creator needs a _very_ specific structure to bring new objects back into CC4.

All objects must be a direct child of the armature.
They each must have an armature modifier which has the armature object set to the characters armature.
And must have weighted vertex groups.

Without these Character Creator will either reject the Fbx import, or get it completely wrong, which is why the add-on won't export it. Otherwise you would just be wasting your time.

To fix it: 
re-parent those 4 objects to the armature.
make sure they each have an armature modifier and the Object is set to the character armature.
each object is vertex weighted to the character. You can copy vertex weights from the Body to the new objects in weight paint mode.

Then it should export and work in CC4.

**Comment by soupday** [comment]
It should have written the .json file with the character are you keeping that file with the .fbx? Also the fbxkey is also copied from the original export (if needed)

it generates 3 files that need to be kept together:
-  _character_name_.fbx
-  _character_name_.json
-  _character_name_.fbxkey
And this is what the plugin in CC4 looks for.

Can you check that these three files are exported from Blender.

**Comment by drkl151** [comment]
Yes, that's right, there is no json file when exporting, what should I do to ensure that it is exported automatically? )

**Comment by soupday** [comment]
Well here's the thing. Even if the .json is missing or corrupted from the original CC4 export. The add-on will create a new one when it exports back to CC4, so there _really_ should be a .json file.

Unless the add-on threw an error mid way through exporting.
Were there any errors generated, in the system console (Window->Toggle System Console) it should show any errors and where it went wrong.

**Comment by MOTO4GIT** [comment]
export all objects you want as obj and scal them to 100, then import them in CC as accessories and make parenting in CC
